### PR TITLE
ANI annotations, dependency updates and pipeline cleanup

### PR DESCRIPTION
**Pull request by crosenth**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,13 @@
 Changes for ya16sdb
 ===================
 
+0.8
```
Comment (dhoogest) [inline CHANGELOG.rst:5]: Looks like 0.7 tag was never actually applied, so perhaps should just merge these changes in with the 0.7 section and plan for that tag to refer to this release.

**Review by dhoogest — APPROVED** [review]

File: bin/refresh.py [hunk 2]
```diff
@@ -204,22 +201,21 @@ def main():
     if any previous records appear again in a future esearch
     we can re-download.
     '''
-    previous_records = set(v.strip() for v in args.previous_records)
-    previous_records = set(v for v in previous_records if v in ncbi)
```
Comment (dhoogest) [inline bin/refresh.py:202]: Fancy - haven't come across `&=` before 😄

File: bin/ani.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        dtype=str,
+        header=None,
+        na_values='na',
+        names=ani_header,
+        sep='\t',
+        skiprows=[0],  # header row
+        usecols=[
+            'assembly_genbank',
+            'assembly_refseq',
+            'best-match-species-name',
```
Comment (dhoogest) [inline bin/ani.py:83]: I think we'll need also the following to meet Mary's spec in https://gitlab.labmed.uw.edu/molmicro/ops/-/issues/502

```
                          'best-match-type-category',
                          'best-match-type-ANI',
                          'best-match-type-qcoverage',
                          'declared-type-ANI',
                          'declared-type-qcoverage'
```

File: bin/ani.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    asm_map = pandas.read_csv(
+        args.asm,
+        dtype=str,
+        header=None,
+        names=asm_header,
+        sep='\t',
+        skiprows=[0, 1],  # comment and header rows
+        usecols=['assembly_genbank', 'wgs_master'])
+    asm_map = asm_map[~asm_map['wgs_master'].isna()]
+    asm_map['wgs_prefix'] = asm_map['wgs_master'].apply(lambda x: x[:7])
```
Comment (dhoogest) [inline bin/ani.py:98]: Can you describe checks to confirm that 'first 7 characters' hold up? I took the somewhat more complicated route to identify the 4/6 alpha prefaces with https://gitlab.labmed.uw.edu/molmicro/reference-packages/plotting/-/merge_requests/10/diffs#0ac059dd44e1a97bdc55b61fb5e9a1d3a762ade9_138_153, but this is simpler if it works reliably.

File: bin/ani.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    info['wgs_prefix'] = info['accession'].apply(lambda x: x[:7])
+    info = info.merge(ani_map, how='left')
+    info = info.merge(asm_map, how='left', on='wgs_prefix', suffixes=['', '_'])
+    info['assembly_genbank'] = info['assembly_genbank'].fillna(
+        info['assembly_genbank_'])
+    drop = ['assembly_genbank_', 'assembly_refseq', 'wgs_prefix', 'wgs_master']
+    info = info.drop(drop, axis='columns')
+    info = info.merge(ani, how='left')
```
Comment (dhoogest) [inline bin/ani.py:106]: May be a little more intuitive for readers to include an `on=` arg here (and in line 100) ... basically just to make more explicit what is going on.

Comment (crosenth) [reply bin/ani.py:106]: Will do, great idea.  

I'll be the first one to admit this code is not clear at all.  Basically I had to get values for a single column 'assembly_genbank' first from the ANI table and then from the ASM file.  I'll see if I can think of some additional tweaks to make it clearer after a week off of not looking at it.

Thanks!

File: bin/ani.py (hunk 4 above)
Comment (crosenth) [reply bin/ani.py:104]: I can dig into this

File: bin/ani.py (hunk 5 above)
Comment (dhoogest) [reply bin/ani.py]: Looks great - thank!

**Pull request merged**